src/Integrations/POSCatalog/AsyncGenerator.php

```diff
@@ -79,38 +74,62 @@ public function init( POSIntegration $integration ) {
 	 */
 	public function register_hooks(): void {
 		add_action( self::FEED_GENERATION_ACTION, [ $this, 'feed_generation_action' ] );
-		add_action( self::FEED_DELETION_ACTION, [ $this, 'feed_deletion_action' ] );
+		add_action( self::FEED_DELETION_ACTION, [ $this, 'feed_deletion_action' ], 10, 2 );
 	}
 
 	/**
 	 * Returns the current feed generation status.
 	 * Initiates one if not already running.
 	 *
-	 * @return array The feed generation status.
+	 * @param array|null $args The arguments to pass to the action.
+	 * @return array           The feed generation status.
 	 */
-	public function get_status(): array {
-		$status = get_option( self::OPTION_KEY );
-
-		if ( false === $status ) {
-			// Clear all previous actions to avoid race conditions.
-			as_unschedule_all_actions( self::FEED_GENERATION_ACTION );
-
-			// Add a bit of delay to avoid race conditions.
-			$delay     = 10;
-			$action_id = as_schedule_single_action( time() + $delay, self::FEED_GENERATION_ACTION, [] );
-
-			$status = [
-				'action_id' => $action_id,
-				'state'     => self::STATE_SCHEDULED,
-				'progress'  => 0,
-				'processed' => 0,
-				'total'     => -1,
-			];
-
-			update_option(
-				self::OPTION_KEY,
-				$status
-			);
+	public function get_status( ?array $args = null ): array {
+		// Determine the option key based on the integration ID and arguments.
+		$option_key = $this->get_option_key( $args );
+		$status     = get_option( $option_key );
+
+		// For existing jobs, make sure that everything in the status makes sense.
+		if ( false !== $status && ! $this->validate_status( $status ) ) {
+			$status = false;
+		}
+
+		// If the status is an array, it means that there is nothing to schedule in this method.
+		if ( false !== $status ) {
+			return $status;
+		}
+
+		// Clear all previous actions to avoid race conditions.
+		as_unschedule_all_actions( self::FEED_GENERATION_ACTION, [ $option_key ], 'wpfoai' );
+
+		$status = [
+			'scheduled_at' => time(),
+			'state'        => self::STATE_SCHEDULED,
+			'progress'     => 0,
+			'processed'    => 0,
+			'total'        => -1,
+			'args'         => $args ?? [],
+		];
+
+		update_option(
+			$option_key,
+			$status
+		);
+
+		// Start an immediate async action to generate the feed.
+		as_enqueue_async_action(
+			self::FEED_GENERATION_ACTION,
+			[ $option_key ],
+			'wpfoai',
+			true,
+			1
+		);
+
+		// Manually force an async request to be dispatched to process the action immediately.
+		if ( class_exists( ActionScheduler_AsyncRequest_QueueRunner::class ) && class_exists( ActionScheduler_Store::class ) ) {
```

Comment:
If always trying to run the feed generation right way, what's the purpose of the async mechanism `ActionScheduler`?